app/src/main/java/org/mifos/mobile/ui/savings_account_transaction/SavingAccountsTransactionScreen.kt

```diff
@@ -0,0 +1,399 @@
+    viewModel.setCheckboxStatesList(context)
+
+    SavingsAccountTransactionScreen(
+        uiState = uiState,
+        viewModel = viewModel,
+        navigateBack = navigateBack,
+        retryConnection = { viewModel.loadSavingsWithAssociations(viewModel.savingsId) },
+        dialogState = dialogState,
+        context = context
+    )
+}
+
+@Composable
+fun SavingsAccountTransactionScreen(
+    uiState: SavingsAccountUiState,
+    viewModel: SavingAccountsTransactionViewModel,
+    navigateBack: () -> Unit,
+    retryConnection: () -> Unit,
+    dialogState: Boolean,
+    context: Context
+) {
+    val savingAccount = rememberSaveable { mutableStateOf(SavingsWithAssociations()) }
+    val transactionList: List<Transactions?> = savingAccount.value.transactions.toList()
+
+    Column(modifier = Modifier.fillMaxSize()) {
+
+        SavingAccountsTransactionTopBar(
+            navigateBack = navigateBack,
+            filterDialog = { viewModel.setDialogOpen(true) }
+        )
+
+        Box(modifier = Modifier.weight(1f)) {
+
+            when (uiState) {
+                is SavingsAccountUiState.SuccessLoadingSavingsWithAssociations -> {
+                    if(uiState.savingAccount.transactions.isNotEmpty()) {
+                        savingAccount.value = uiState.savingAccount
+                        SavingsAccountTransactionContent(transactionList = transactionList)
+                    } else {
+                        MifosErrorComponent(isEmptyData = true)
+                    }
+                }
+
+                is SavingsAccountUiState.Loading -> {
+                    MifosProgressIndicatorOverlay()
+                }
+
+                is SavingsAccountUiState.Error -> {
+                    MifosErrorComponent(
+                        isNetworkConnected = Network.isConnected(context),
+                        isEmptyData = false,
+                        isRetryEnabled = true,
+                        onRetry = retryConnection
+                    )
+                }
+
+                is SavingsAccountUiState.ShowFilteredTransactionsList -> {
+                    if(uiState.savingAccountsTransactionList != null) {
+                        if(uiState.savingAccountsTransactionList.isNotEmpty()) {
+                            SavingsAccountTransactionContent(
+                                transactionList = uiState.savingAccountsTransactionList
+                            )
+                        }else {
+                            EmptyDataComponentWithModifiedMessageAndIcon(isEmptyData = true,
+                                message = stringResource(id = R.string.no_transaction_found),
+                                icon = Icons.Filled.CompareArrows)
+                        }
+                    } else {
+                        MifosErrorComponent(isEmptyData = true)
+                    }
+                }
+
+                is SavingsAccountUiState.Initial -> Unit
+
+                else -> Unit
```

Comment:
no need of else. make a new SavingsAccountTransactionUiState in viewmodel specific to only this screen.